vendor/src/github.com/vbatts/tar-split/tar/asm/assemble.go

```diff
@@ -0,0 +1,68 @@
+package asm
+
+import (
+	"bytes"
+	"fmt"
+	"hash/crc64"
+	"io"
+
+	"github.com/vbatts/tar-split/tar/storage"
+)
+
+// NewOutputTarStream returns an io.ReadCloser that is an assemble tar archive
+// stream.
+//
+// It takes a storage.FileGetter, for mapping the file payloads that are to be read in,
+// and a storage.Unpacker, which has access to the rawbytes and file order
+// metadata. With the combination of these two items, a precise assembled Tar
+// archive is possible.
+func NewOutputTarStream(fg storage.FileGetter, up storage.Unpacker) io.ReadCloser {
+	// ... Since these are interfaces, this is possible, so let's not have a nil pointer
+	if fg == nil || up == nil {
+		return nil
+	}
+	pr, pw := io.Pipe()
+	go func() {
+		for {
+			entry, err := up.Next()
+			if err != nil {
+				pw.CloseWithError(err)
+				break
+			}
+			switch entry.Type {
+			case storage.SegmentType:
+				if _, err := pw.Write(entry.Payload); err != nil {
+					pw.CloseWithError(err)
+					break
+				}
+			case storage.FileType:
+				if entry.Size == 0 {
+					continue
+				}
+				fh, err := fg.Get(entry.Name)
+				if err != nil {
+					pw.CloseWithError(err)
+					break
+				}
+				c := crc64.New(storage.CRCTable)
+				tRdr := io.TeeReader(fh, c)
+				if _, err := io.Copy(pw, tRdr); err != nil {
+					fh.Close()
+					pw.CloseWithError(err)
+					break
+				}
+				if !bytes.Equal(c.Sum(nil), entry.Payload) {
```

Comment:
Just to clarify my doubts here, I tried the following:
- pull down image `alpine`
- tamper with the data on disk (modified `/var/lib/docker/graph/<id>/root/etc/passwd`)
- push image back as `icecrimetest/alpine`

What I see is a successful push of what I'm assuming is an inconsistent image (data / metadata mismatch), while I was assuming that test would make it break. What am I missing?